src/Resources/config/debug.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+use Kreyu\Bundle\DataTableBundle\DataCollector\DataTableDataCollector;
+use Kreyu\Bundle\DataTableBundle\Debug\TraceableDataTableFactory;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('debug.kreyu_data_table.factory', TraceableDataTableFactory::class)
```

Comment:
To keep with the practices of prefixing the services with bundle name, I suggest naming this `kreyu_data_table.debug.factory` (consistent with the data collector service name suggested below)